Server/src/services/dataExtraction/JsonFileHandlerFactory.ts

```diff
@@ -14,11 +15,23 @@ export class JsonFileExtractorFactory extends FileExtractorFactory {
 }
 
 export class JsonFileExtractor extends AbstractFileExtractor {
+    private categoryModel: CategoryModel
 
-    parseFile(): Partial<IDataSetModel> {
+    constructor(filePath: string) {
+        super(filePath);
+        this.categoryModel = new CategoryModel()
+    }
+    async parseFile(): Promise<Partial<IDataSetModel>> {
         try {
-
             let parsedFileData: Partial<IDataSetModel> = JSON.parse(fileSystem.readFileSync(this.filePath))
+            let categoryID = await this.categoryModel.findCategoryIDFromSubcategoryID(parsedFileData.subcategory)
```

Comment:
I think this approach might be a little off. We can't expect users to know what ID to enter for a subcategory.

For example: How will a user know what subcategory ID corresponds to cell width? It makes more sense for them to enter a string and we match the string to our list

Reply:
To my understanding the file upload was meant to in part be similar to what is returned to a user, as we return IDs not names. It also follows the recent units/dimensions change which does the same thing. However, I believe you bring up a good point that justifies a redo of how both are handled. I have an idea on how to do for category/subcategory, I'll give a go at implementing it.

Reply:
This portion is what's returned to the user... But after they upload a file. When they are uploading a file they won't know which id maps to which name. Now that I think about it, it might be better just not to accept a category/subcategory and let the user input the category/subcategory from the dataset dropdown